File: 02_ProgrammingFundamentals/_03_Arrays/_01_Lecture/_07_CondenseArrayToNumber.java
package _02_ProgrammingFundamentals._03_Arrays._01_Lecture;

import java.util.Arrays;
import java.util.Scanner;

public class _07_CondenseArrayToNumber {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Raed input values
        int[] numbers = Arrays.stream(scan.nextLine().split("\\s+")).mapToInt(Integer::parseInt).toArray();

        // Condense the array
        while (numbers.length > 1) {
            int[] condensed = new int[numbers.length - 1];

            for (int i = 0; i < numbers.length - 1; i++) {
                condensed[i] = numbers[i] + numbers[i + 1];
            }

            numbers = condensed;
        }

        // Print result
        System.out.println(numbers[0]);

        scan.close();
    }
}

File: 24_C#Fundamentals/_03_Arrays/_01_Lecture/_07_CondenseArrayToNumber/_07_CondenseArrayToNumber.cs
using System;

public class _07_CondenseArrayToNumber
{
    public static void Main(string[] args)
    {
        // Raed input values
        int[] numbers = Console.ReadLine().Split().Select(int.Parse).ToArray();

        // Condense the array
        while (numbers.Length > 1)
        {
            int[] condensed = new int[numbers.Length - 1];

            for (int i = 0; i < numbers.Length - 1; i++)
            {
                condensed[i] = numbers[i] + numbers[i + 1];

            }

            numbers = condensed;
        }

        // Print result
        Console.WriteLine(numbers[0]);
    }
}
